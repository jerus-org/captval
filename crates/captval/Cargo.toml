[package]
name = "captval"
version = "0.1.0"
description = "Captcha validators"
license = "MIT"
documentation = "https://docs.rs/captval"
repository = "https://github.com/jerus-org/captval"
edition.workspace = true
rust-version.workspace = true
publish = true

[dependencies]
captval_derive = { path = "../captval_derive" }
hex = { workspace = true, optional = true }
reqwest.workspace = true
serde.workspace = true
serde_json.workspace = true
serde_urlencoded.workspace = true
thiserror.workspace = true
tracing = { workspace = true, optional = true }
url.workspace = true
uuid.workspace = true

[dev-dependencies]
chrono.workspace = true
claims.workspace = true
env_logger.workspace = true
itertools.workspace = true
lambda_runtime.workspace = true
mockd = { workspace = true }
rand.workspace = true
serde_test.workspace = true
tokio.workspace = true
tracing-appender.workspace = true
tracing-bunyan-formatter.workspace = true
tracing-log.workspace = true
tracing-subscriber.workspace = true
tracing-test.workspace = true
trybuild.workspace = true
wiremock.workspace = true

[features]
default = ["hcaptcha", "ext", "rustls-backend"]
enterprise = []
ext = ["dep:hex"]
hcaptcha = []
nativetls-backend = ["reqwest/native-tls"]
nightly = []
rustls-backend = ["reqwest/rustls-tls"]
trace = ["dep:tracing"]
