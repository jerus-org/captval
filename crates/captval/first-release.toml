pre-release-commit-message = "chore: Release v0.1.0"
tag-message = "{{tag_name}}"
tag-name = "{{prefix}}v{{version}}"
sign-tag = true
sign-commit = true
consolidate-commits = true
allow-branch = ["main"]
pre-release-replacements = [
    { file = "CHANGELOG.md", search = "## \\[Unreleased\\]", replace = "## [{{version}}] - {{date}}", exactly = 1 },
]
