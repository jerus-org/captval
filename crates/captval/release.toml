pre-release-commit-message = "chore: Release {{crate_name}} {{version}}"
tag-message = "{{tag_name}}"
tag-name = "{{prefix}}v{{version}}"
consolidate-commits = false
allow-branch = ["main"]
sign-commit = true
sign-tag = true
pre-release-replacements = [
    { file = "CHANGELOG.md", search = "## \\[Unreleased\\]", replace = "## [{{version}}] - {{date}}", exactly = 1 },
    { file = "CHANGELOG.md", search = "\\[Unreleased\\]:", replace = "[{{version}}]:", exactly = 1 },
    { file = "CHANGELOG.md", search = "\\.\\.\\.HEAD", replace = "...{{tag_name}}", exactly = 1 },
    { file = "README.md", search = "hcaptcha = .*", replace = "{{crate_name}} = \"{{version}}\"" },
    { file = "src/lib.rs", search = "version = .+,", replace = "version = \"{{version}}\",", exactly = 1 },
    { file = "sonar-project.properties", search = "sonar.projectVersion=.+", replace = "sonar.projectVersion=\"{{version}}\",", exactly = 1 },
]
