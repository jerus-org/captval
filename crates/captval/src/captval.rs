//! Captval Trait
use crate::{Error, Response};
// use async_trait::async_trait;
use std::{future::Future, pin::Pin};

/// Captval trait
pub trait Captval {
    /// valid response function
    fn valid_response(
        &self,
        secret: &str,
        uri: Option<String>,
    ) -> Pin<Box<dyn Future<Output = Result<Response, Error>>>>;
}
