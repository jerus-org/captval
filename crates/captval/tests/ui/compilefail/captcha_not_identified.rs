use captval::Captval;

#[derive(Captval)]
struct Test {
    Captval: String,
}

fn main() {
    println!("Super!");
}
