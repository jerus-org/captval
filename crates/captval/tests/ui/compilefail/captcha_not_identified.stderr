error: Field containing captval not identified

         = help: The field containing the captval response string must be identified with #[captcha]

               #[derive(Captval)]
               struct MyStruct {
                   #[captcha]
                   captval: String,
               }

 --> tests/ui/compilefail/captcha_not_identified.rs:4:8
  |
4 | struct Test {
  |        ^^^^
